### DISCUSSION TITLE
Button 2.0 BETA with component properties

### DISCUSSION THREAD

**Discussion by clastzoo** [baloise/design-system#1006] [Figma Updates]
Hi Baloise Design System and Figma Friends

We would like to inform you that the new Button Component (Naming: **Button-[BETA]**) is now available in the component library. We recently restructured the architecture by using component properties instead of master buttons. Therefore, we would like to test it first to ensure that everything works as expected.

You are encouraged to use the new button in all your files from now on. After a successful test period of approximately one month (until the end of August), the old button will be deprecated.

Don't worry about your old designs as the change does not affect them. After we roll out the button definitively, the only thing you need to do is manually update the old buttons to the new one.

To accomplish this, use `cmd + /`  to bring up the search bar and enter the search term **select all with same instance**. From there, you can select the "New" button (See video below). Although the label text within the button will be preserved, the rest of the styling will need to be manually adjusted due to the architecture change.

https://github.com/baloise/design-system/assets/62377419/e8654a02-d7ed-4553-9687-9adc37748e30

**Button properties**

![button-component-properties](https://github.com/baloise/design-system/assets/62377419/1618deea-dd09-4280-936f-9aa3a6f55099)

**Type**
Primary
Secondary
Tertiary (was Text button previously)

- Default
- Purple
- Green
- Red
- Orange

Link
Success
Warning
Danger

**Size:** 
Medium (Default)
Small

**State:**
Default
Hover
Active
Focus
Loading
Disabled

**Themed (Tertiary only)**
Default (Tertiary)

**Icon Button**
Yes / No
Left
Right

**Label**

**Next Steps:**

Please give us feedback if anything is not working as expected, or you have suggestions. We will keep you posted and let you know when the button is out of beta stadium. 

From a technical point of view the buttons are ready to use, you find the documentations here:
https://design.baloise.dev/?path=/docs/components-button--basic

**Comment by joachimprinzbach** [comment]
whoop

**Reply by joachimprinzbach** [reply]
reply to this!

**Comment by joachimprinzbach** [comment]
> I just want to test this new feature.

quoted!
